core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1260,6 +1260,14 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
           ReturnTypes.IDENTITY_TO_MAP_NULLABLE,
           OperandTypes.STRING_OPTIONAL_STRING_OPTIONAL_STRING);
 
+  /** The "SUBSTRING_INDEX(string, delimiter, count)" function. */
+  @LibraryOperator(libraries = {SPARK})
+  public static final SqlFunction SUBSTRING_INDEX =
+      SqlBasicFunction.create(SqlKind.SUBSTRING_INDEX,
+          ReturnTypes.ARG0_NULLABLE_VARYING,
```

Comment:
If any of the argument is null, this returns null. 
I'm not very familiar with Spark, does this behave like Spark?

Reply:
Yes，I ran some cases on spark and confirmed that if any of the argument is null, this returns null.

```
select substring_index('abc',cast(null as varchar(1)), cast(null as integer));
+--------------------------------+
| substring_index(abc, CAST(NULL |
| AS STRING), CAST(NULL AS INT)) |
+--------------------------------+
| null                           |
+--------------------------------+


 select substring_index('abc',cast(null as varchar(1)), 1);
+--------------------------------+
| substring_index(abc, CAST(NULL |
|         AS STRING), 1)         |
+--------------------------------+
| null                           |
+--------------------------------+


select substring_index(cast(null as varchar(3)),'.', 1);
+--------------------------------+
|  substring_index(CAST(NULL AS  |
|         STRING), ., 1)         |
+--------------------------------+
| null                           |
+--------------------------------+
```

Reply:
Thanks for your explanation.